src/realm/parser/driver.cpp

```diff
@@ -931,9 +931,23 @@ std::unique_ptr<Subexpr> ConstantNode::visit(ParserDriver* drv, DataType hint)
                     case type_Float:
                         ret = new Value<float>(drv->m_args.float_for_argument(arg_no));
                         break;
-                    case type_Double:
-                        ret = new Value<double>(drv->m_args.double_for_argument(arg_no));
+                    case type_Double: {
```

Comment:
I don't understand why this is required. I thought we would want to keep the double as it is so that the matching uses the special mixed comparisons between numeric types instead of simple casting here.

Reply:
Yes, but in realm-js all number type arguments are returned as double. If we did not do this, we would miss the option to use the optimized node instead of the general Compare<T> class. So this is an optimization.